File: SocialAppFlutter/social_app/lib/Models/event.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/location.dart';
import 'package:social_app/Models/sport_category.dart';
import 'package:social_app/Models/user.dart';

part 'json_generated_code/event.g.dart';

@JsonSerializable(explicitToJson: true)
class Event {
  final int id;
  final DateTime startDateTime;
  final double duration;
  final SportCategory sportCategory;
  final Location location;
  final User creator;
  final int requiredMembersTotal;

  Event(this.id, this.startDateTime, this.duration, this.sportCategory,
      this.location, this.creator, this.requiredMembersTotal);

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);

  Map<String, dynamic> toJson() => _$EventToJson(this);
}

File: APIs/ChatService/ChatService/Models/Event.cs
using System;
using System.Collections.Generic;

namespace ChatService.Models
{
    public partial class Event
    {
        public Event()
        {
            EventMembers = new HashSet<EventMember>();
            GoogleCalendarEvents = new HashSet<GoogleCalendarEvent>();
            Honors = new HashSet<Honor>();
            Invitations = new HashSet<Invitation>();
            RatedEvents = new HashSet<RatedEvent>();
        }

        public int Id { get; set; }
        public DateTime StartDatetime { get; set; }
        public double Duration { get; set; }
        public int SportCategory { get; set; }
        public int LocationId { get; set; }
        public int CreatorId { get; set; }
        public int RequiredMembersTotal { get; set; }

        public virtual User Creator { get; set; } = null!;
        public virtual Location Location { get; set; } = null!;
        public virtual SportCategory SportCategoryNavigation { get; set; } = null!;
        public virtual ICollection<EventMember> EventMembers { get; set; }
        public virtual ICollection<GoogleCalendarEvent> GoogleCalendarEvents { get; set; }
        public virtual ICollection<Honor> Honors { get; set; }
        public virtual ICollection<Invitation> Invitations { get; set; }
        public virtual ICollection<RatedEvent> RatedEvents { get; set; }
    }
}
